backend/src/test/java/net/vidux/camhub/discovery/DiscoveryServiceTest.java

```diff
@@ -0,0 +1,58 @@
+package net.vidux.camhub.discovery;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class DiscoveryServiceTest {
+
+  @Mock CompletableFuture<Set<RawCameraData>> cameraScanTask;
+
+  @InjectMocks @Spy DiscoveryService spiedDiscover;
+
+  @Test
+  void requestDiscoverTestWhenScanIsRunning() {
+    when(cameraScanTask.isDone())
+        .thenAnswer(
+            invocation -> {
+              return false;
+            });
+
+    Assertions.assertThrows(DiscoveryException.class, spiedDiscover::requestDiscovery);
+  }
+
+  @Test
+  void requestDiscoverTestWhenNoScanIsRunning() {
+    doNothing().when(spiedDiscover).discover();
```

Comment:
Szerintem nem kéne, hogy spy legyen a `DiscoveryService`.

Inkább a `CameraScan` component-et kellene mock-olni és annak a `cameraScanTask()` metódusának kellene megadni, hogy egy mock-olt `CompletableFuture`-t adjon vissza: 
`when(mockedCameraScan.cameraScanTask().thenReturn(CompletableFuture.completedFuture(Set.of(...)));`

Mivel így a `publishRawCameraData` metódus is meg lesz hívva, ezért a `RawCameraEventPublisher` osztályt is mock-olni kellene, de ennek viszont nem kell metódusait mock-olni, mert a mock osztályok void metódusa nem csinál semmit se.